components/widgets/settings/text_setting.tsx

```diff
@@ -1,73 +1,65 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
-import Setting from './setting.jsx';
+import Setting from './setting';
 
-export default class TextSetting extends React.Component {
-    static validTypes = [
-        'input',
-        'textarea',
-        'number',
-        'email',
-        'tel',
-        'url',
-        'password',
-    ];
-    static propTypes = {
-        id: PropTypes.string.isRequired,
-        label: PropTypes.node.isRequired,
-        labelClassName: PropTypes.string,
-        placeholder: PropTypes.string,
-        helpText: PropTypes.node,
-        footer: PropTypes.node,
-        value: PropTypes.oneOfType([
-            PropTypes.string,
-            PropTypes.number,
-        ]).isRequired,
-        inputClassName: PropTypes.string,
-        maxLength: PropTypes.number,
-        resizable: PropTypes.bool,
-        onChange: PropTypes.func,
-        disabled: PropTypes.bool,
-        type: PropTypes.oneOf(TextSetting.validTypes),
-    };
+type InputTypes = 'input' |'textarea'| 'number'| 'email'| 'tel'| 'url'| 'password'
```

Comment:
Could you please fix spacing between types? (I wish linting can help with it but apparently not.)

Reply:
oh yes i will change this,
Also any specific reason for not running prettier, i am sure it could have formatted those.

Reply:
We looked into it before and there were format/styles that's different from what we set in our standard style.